Backport patch to remove deprecation warning from upstream commit
778ec7f43cd1e8de272c2fc7023ce9d52fa74f9d.

Index: lib/capybara/webkit/node.rb
--- lib/capybara/webkit/node.rb.orig
+++ lib/capybara/webkit/node.rb
@@ -6,12 +6,30 @@ module Capybara::Webkit
     end
 
     def visible_text
-      Capybara::Helpers.normalize_whitespace(invoke("text"))
+      text = invoke(:text).to_s
+      if Capybara::VERSION.to_f < 3.0
+        Capybara::Helpers.normalize_whitespace(text)
+      else
+        text.gsub(/\ +/, ' ')
+            .gsub(/[\ \n]*\n[\ \n]*/, "\n")
+            .gsub(/\A[[:space:]&&[^\u00a0]]+/, "")
+            .gsub(/[[:space:]&&[^\u00a0]]+\z/, "")
+            .tr("\u00a0", ' ')
+      end
     end
     alias_method :text, :visible_text
 
     def all_text
-      Capybara::Helpers.normalize_whitespace(invoke("allText"))
+      text = invoke(:allText)
+      if Capybara::VERSION.to_f < 3.0
+        Capybara::Helpers.normalize_whitespace(text)
+      else
+        text.gsub(/[\u200b\u200e\u200f]/, '')
+            .gsub(/[\ \n\f\t\v\u2028\u2029]+/, ' ')
+            .gsub(/\A[[:space:]&&[^\u00a0]]+/, "")
+            .gsub(/[[:space:]&&[^\u00a0]]+\z/, "")
+            .tr("\u00a0", ' ')
+      end
     end
 
     def [](name)
